Resources/doc/operations.md

```diff
@@ -0,0 +1,99 @@
+methods.
+
+The following `Resource` definition exposes a `GET` operation for it's collection but not the `POST` one:
+
+```yaml
+services:
+    resource.product.collection_operation.get:
+        class:     "Dunglas\ApiBundle\Api\Operation\Operation"
+        public:    false
+        factory:   [ "@api.operation_factory", "createItemOperation" ]
+        arguments: [ "@resource.product", "GET" ]
+
+    resource.product:
+        parent:    "api.resource"
+        arguments: [ "AppBundle\Entity\Product" ]
+            -      [ "addCollectionOperation", [ "@resource.product.collection_operation.get" ] ]
+        tags:      [ { name: "api.resource" } ]
+```
+
+## Creating custom operations
+
+Sometimes, it can be useful to [create custom controller](8-custom-controllers.md) actions. DunglasApiBundle allows to register custom operations
+for both collections and items. It will register them automatically in the Symfony routing system and will expose them in
+the Hydra vocab (if enabled).
+
+```yaml
+    resource.product.item_operation.get:
+        class:     "Dunglas\ApiBundle\Api\Operation\Operation"
+        public:    false
+        factory:   [ "@api.operation_factory", "createItemOperation" ]
+        arguments: [ "@resource.product", "GET" ]
+
+    resource.product.item_operation.put:
+        class:     "Dunglas\ApiBundle\Api\Operation\Operation"
+        public:    false
+        factory:   [ "@api.operation_factory", "createItemOperation" ]
+        arguments: [ "@resource.product", "PUT" ]
+
+
+    resource.product.item_operation.custom_get:
+        class:   "Dunglas\ApiBundle\Api\Operation\Operation"
+        public:  false
+        factory: [ "@api.operation_factory", "createItemOperation" ]
+        arguments:
+            -    "@resource.product"               # Resource
+            -    [ "GET", "HEAD" ]                 # Methods
+            -    "/products/{id}/custom" # Path
+            -    "AppBundle:Custom:custom"         # Controller
+            -    "my_custom_route"                 # Route name
+            -    # Context (will be present in Hydra documentation)
+                 "@type":       "hydra:Operation"
+                 "hydra:title": "A custom operation"
+                 "returns":     "xmls:string"
+
+    resource.product:
+        parent:    "api.resource"
+        arguments: [ "AppBundle\Entity\Product" ]
+        calls:
+            -      method:    "addItemOperation"
+                   arguments:
+                       -      "@resource.product.item_operation.get"
+            -      method:    "addItemOperation"
+                   arguments:
+                       -      "@resource.product.item_operation.put"
+            -      method:    "addItemOperation"
+                   arguments:
+                       -      "@resource.product.item_operation.custom_get"
```

Comment:
Wouldn't it be possible to pass multiple operations via `addItemOperation`? That would result in:

```yml
- method:    "addItemOperation"
  arguments:
      -      "@resource.product.item_operation.get"
      -      "@resource.product.item_operation.put"
      -      "@resource.product.item_operation.custom_get"
```